src/control/lib/netdetect/netdetect.go

```diff
@@ -0,0 +1,163 @@
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package netdetect
+/*
+#cgo CFLAGS: -I${SRCDIR}/../../include
+#cgo LDFLAGS: -lhwloc
+#include <stdlib.h>
+#include <hwloc.h>
+#include <stdio.h>
+*/
+import "C"
+import "unsafe"
+
+import (
+	"github.com/daos-stack/daos/src/control/log"
+	"github.com/pkg/errors"
+	"net"
+	"strings"
+)
+
+// initLib initializes the hwloc library.  For testing purposes,
+// set xmlTopology to a valid XML topology file name (generated by lstopo)
+// and hwloc will make queries against the data in this file.  Otherwise,
+// specify no topology file and hwloc will perform queries against the actual
+// local system.
+func initLib(xmlTopology string) (C.hwloc_topology_t, error) {
+	var topology C.hwloc_topology_t
+
+	status := C.hwloc_topology_init(&topology)
+	if status != 0 {
+		return nil, errors.New("hwloc_topology_init failure")
+	}
+
+	if (xmlTopology != "") {
+		status := C.hwloc_topology_set_xml(topology, C.CString(xmlTopology))
+		if (status != 0) {
+			return nil, errors.New("hwloc_topology_set_xml failure")
+		}
+	}
+
+	status = C.hwloc_topology_set_flags(topology, C.HWLOC_TOPOLOGY_FLAG_IO_DEVICES)
+	if status != 0 {
+		return nil, errors.New("hwloc_topology_set_flags failure")
+	}
+
+	status = C.hwloc_topology_load(topology)
+	if status != 0 {
+		return nil, errors.New("hwloc_topology_load failure")
+	}
+	return topology, nil
+}
+
+// cleanUp closes out the hwloc resources
+func cleanUp(topology C.hwloc_topology_t) {
+	if (topology != nil) {
+		C.hwloc_topology_destroy(topology)
+	}
+}
+
+// GetAffinityForNetworkDevices searches the system topology for each network
+// device specified by the input string and returns corresponding cpuset and
+// nodeset bitmaps as strings.  When converted back to hwloc_bitmap_t via
+// hwloc_bitmap_sscanf() these bitmaps are used by the hwloc API to bind a
+// thread to processing units that are closest to the given network device.
+//
+// The input netNames string has the form "device0,device1,...deviceN"
+// where 'device' specifies a network device such as "eth0", "eth1", or "ib0"
+// The return string has the form:
+// "dev0:cpuset:nodeset;dev1:cpuset:nodeset; ... devN:cpuset:nodeset"
+// where dev is the name of the network device and cpuset and nodeset are
+// the corresponding bitmap strings.
+//
+// Network device names that are not found in the topology are ignored.
+// The order of network devices in the return string depends on the natural
+// order in the system topology and does not depend on the order specified
+// by the input string.
+func GetAffinityForNetworkDevices(netNames string,
+	xmlTopology string) (string, error) {
```

Comment:
OK, I see what you were talking about in terms of loading test fixtures now. I'm not crazy about exposing this as a required parameter on a public API function.

Here's what I propose instead: Remove everything in netdetect.go concerned with loading test fixtures. Make a small change in netdetect_test.go, like so:
```go
package netdetect_test

import (
	"os"
	"testing"

	"github.com/daos-stack/daos/build/src/control/src/github.com/daos-stack/daos/src/control/lib/netdetect"
	. "github.com/daos-stack/daos/src/control/common"
	"github.com/daos-stack/daos/src/control/log"
)

func withTopologyFile(xmlTopology string, testFunc func(string) (string, error), deviceString string) (string, error) {
	if _, err := os.Stat(xmlTopology); err != nil {
		return "", err
	}

	os.Setenv("HWLOC_XMLFILE", xmlTopology)
	defer os.Unsetenv("HWLOC_XMLFILE")

	return testFunc(deviceString)
}

func TestParseTopology(t *testing.T) {
...
	for _, tt := range tests {
		netAdapterAffinity, err := withTopologyFile(tt.topology, netdetect.GetAffinityForNetworkDevices, tt.netDevsList)
		if err != nil {
			log.Debugf("error from GetAffinityForNetworkDevices() %v", err)
		}
		AssertEqual(t, netAdapterAffinity, tt.expected,
			"unexpected mismatch with device and topology")
}
```

This way we keep the messiness contained in the testing package. Sorry about going back and forth on using the hwloc API vs env vars -- I didn't fully understand how it had to work.